### ISSUE TITLE
npm install of 3.19.0 failing

### ISSUE THREAD

**Issue by cf512** [curlconverter/curlconverter#301]
With the recent release, I am not currently able to install curlconverter:

```
➜  testtest git:(master) ✗ npm -v
7.24.0
➜  testtest git:(master) ✗ node -v
v16.10.0
➜  testtest git:(master) ✗ npm install --save-dev curlconverter
npm ERR! code 1
npm ERR! git dep preparation failed
npm ERR! command /usr/local/Cellar/node/16.10.0/bin/node /usr/local/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/Users/omitted/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit --include=dev --include=peer --include=optional --no-package-lock-only --no-dry-run
npm ERR! > yargs-parser@20.2.7 prepare
npm ERR! > npm run compile
npm ERR!
npm ERR!
npm ERR! > yargs-parser@20.2.7 precompile
npm ERR! > rimraf build
npm ERR!
npm ERR!
npm ERR! > yargs-parser@20.2.7 compile
npm ERR! > tsc
npm ERR!
npm ERR! lib/yargs-parser.ts:737:17 - error TS2571: Object is of type 'unknown'.
npm ERR!
npm ERR! 737             if (ex.name === 'PermissionDenied') error = ex
npm ERR!                     ~~
npm ERR!
npm ERR! lib/yargs-parser.ts:737:49 - error [... etc ... etc]
```

**Comment by verhovsky** [comment]
The recent releases (with the exception of 4.0.0-alpha.2, which I fixed after a day or two) have been pre-releases. When you `npm install curlconverter` you're installing  3.19.0. If you want to install the recent release, you have to do 

    npm install curlconverter@next

But this has something to do with 

https://github.com/NickCarneiro/curlconverter/blob/d8c131a4e11b09ce57642bbff4efeb694461ac92/package-lock.json#L17

@NickCarneiro if you rebase that repo and https://github.com/NickCarneiro/yargs-parser-ansi when you get the chance to their upstreams it might solve it. Also, this time around you can skip adding my yargs-parser changes because they don't actually add ANSI-C strings to yargs (as we discussed). idonava's changes are the only changes actually doing anything.

**Comment by cf512** [comment]
``npm install curlconverter@next``
``npm install --save-dev curlconverter@4.0.0-alpha.6``
and even just: ``npm install --save-dev tree-sitter``
all failing with same error:

```
npm ERR! code 1
npm ERR! path /Users/connor/code/testtest/node_modules/tree-sitter
npm ERR! command failed
npm ERR! command sh -c prebuild-install || node-gyp rebuild
npm ERR! CC(target) Release/obj.target/tree_sitter/vendor/tree-sitter/lib/src/lib.o
npm ERR!   LIBTOOL-STATIC Release/tree_sitter.a
npm ERR!   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/binding.o
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@16.10.0 | darwin | x64
npm ERR! gyp info find Python using Python version 3.8.2 found at [... etc ... etc]
```

I wonder if this could have anything to do with me using zsh instead of bash...

**Comment by verhovsky** [comment]
@cf512 no, you're running into https://github.com/tree-sitter/node-tree-sitter/issues/82 . tree-sitter is currently broken on Node 16. [The fix](https://github.com/tree-sitter/node-tree-sitter/pull/92) has been merged, it just needs to be released.

**Comment by dmvict** [comment]
@verhovsky
Thanks for the answer.

Sorry, I do not see official contributors.

**Comment by verhovsky** [comment]
@dmvict I just published 4.0.0-alpha.7 which should have the fix. 

Just to be clear, this is unrelated to the issue described in the first comment though. This only applies if you're installing curlconverter with `npm install curlconverter@next` because curlconverter 4 is still in development. curlconverter 3.19.0 (which is what npm tries to install when you don't include `@next`) is still broken.

**Comment by cf512** [comment]
@verhovsky Yes, it works now. Many thanks.
